### PR TITLE
hysteria: init at 1.1.0

### PR DESCRIPTION
**Pull request by oluceps**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/hysteria/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  lib,
```
Comment (ocfox) [inline pkgs/tools/networking/hysteria/default.nix:2]:
```suggestion
{ lib,
```

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -9633,6 +9633,12 @@
     githubId = 1237862;
     name = "Ollie Bunting";
   };
+  oluceps = {
```
Comment (ocfox) [inline maintainers/maintainer-list.nix:9636]: Please extract this change to a separate commit, see #182036 for an example.

**Review by ocfox — CHANGES_REQUESTED** [review]

**Comment by ocfox** [comment]
Please swap the order of these commits, adding maintainers should be done before the other commit.

File: pkgs/tools/networking/hysteria/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "HyNetwork";
+    repo = "${pname}";
+    rev = "v${version}";
+    sha256 = "sha256-V+umf7+qRANSNsMrU1Vij3ni6ayq/d41xSy3o+7sEHQ=";
+  };
+
+  vendorSha256 = "sha256-oxCZ4+E3kffHr8ca9BKCSYcSWQ8jwpzrFs0fvCvZyJE=";
+  proxyVendor = true;
+
+  doCheck = false;
```
Comment (ocfox) [inline pkgs/tools/networking/hysteria/default.nix:19]: Maybe you should add an explanation for why you skipped the check.

Comment (oluceps) [reply pkgs/tools/networking/hysteria/default.nix:20]: Reason added.

**Review by ocfox — APPROVED** [review]
LGTM

File: pkgs/tools/networking/hysteria/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{ lib,
+  fetchFromGitHub,
+  buildGoModule,
+}:
```
Comment (ocfox) [inline pkgs/tools/networking/hysteria/default.nix:4]:
```suggestion
{ lib
, fetchFromGitHub
, buildGoModule
}:
```

File: pkgs/tools/networking/hysteria/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  vendorSha256 = "sha256-oxCZ4+E3kffHr8ca9BKCSYcSWQ8jwpzrFs0fvCvZyJE=";
+  proxyVendor = true;
+
+  # Network required
+  doCheck = false;
+
+  meta = with lib; {
+    description = "feature-packed proxy & relay utility optimized for lossy, unstable connections";
```
Comment (ocfox) [inline pkgs/tools/networking/hysteria/default.nix:23]:
```suggestion
    description = "A feature-packed proxy & relay utility";
```

**Review by ocfox — CHANGES_REQUESTED** [review]
A few small problems, otherwise it's all good.

**Review by ocfox — APPROVED** [review]

File: pkgs/tools/networking/hysteria/default.nix (hunk 5 above)
Comment (winterqt) [reply pkgs/tools/networking/hysteria/default.nix]: What's the motivation for this change? I know the guidelines say keep it short, but I'd say the previous version is also pretty short...

Comment (ocfox) [reply pkgs/tools/networking/hysteria/default.nix]: Feels that the description of the second half is superfluous, not because of length. But this suggestion should be my fault.

Comment (winterqt) [reply pkgs/tools/networking/hysteria/default.nix]: Since the description comes from upstream, and is short enough, I think it's fine.

For example, look at how it renders on https://search.nixos.org (obviously I edited the page to show this in comparison to another package):

image

Comment (ocfox) [reply pkgs/tools/networking/hysteria/default.nix]: @oluceps, I agree with him. Sorry for my previous suggestion.

Comment (oluceps) [reply pkgs/tools/networking/hysteria/default.nix]: Modified. thx

File: pkgs/tools/networking/hysteria/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+, fetchFromGitHub
+, buildGoModule
+}:
+buildGoModule rec {
+  pname = "hysteria";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "HyNetwork";
+    repo = "${pname}";
```
Comment (Mindavi) [inline pkgs/tools/networking/hysteria/default.nix:11]: I prefer the name and repo to be put here so overriding pname can be done without breakage. But yeah, I know there's bad examples around

**Comment by Mindavi** [comment]
Other than that minor comment, LGTM

File: pkgs/tools/networking/hysteria/default.nix (hunk 6 above)
Comment (oluceps) [reply pkgs/tools/networking/hysteria/default.nix:11]:
```nix
  pname = "hysteria";
  version = "1.1.0";

  src = fetchFromGitHub {
    owner = "HyNetwork";
    repo = "Hysteria";
    rev = "v${version}";
    sha256 = "sha256-V+umf7+qRANSNsMrU1Vij3ni6ayq/d41xSy3o+7sEHQ=";
  };
```
Is this the correct modification?

Comment (Mindavi) [reply pkgs/tools/networking/hysteria/default.nix:11]: Yeah, that looks good to me,

**Review by Mindavi — APPROVED** [review]
Waiting for ofborg...

File: pkgs/tools/networking/hysteria/default.nix (hunk 6 above)
Comment (winterqt) [reply pkgs/tools/networking/hysteria/default.nix]: @Mindavi Overriding `pnane` in this case is fine (and preferred, AFAIK, to avoid repetition) since it's not going off of the final value of `pname`, only the initial value.

If this was using overlay-style `mkDerivation`, you'd be correct. But that's not the case here.
